WordPress/src/main/java/org/wordpress/android/ui/posts/reactnative/ReactNativeRequestHandler.kt

```diff
@@ -0,0 +1,69 @@
+package org.wordpress.android.ui.posts.reactnative
+
+import androidx.core.util.Consumer
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.ReactNativeFetchResponse
+import org.wordpress.android.fluxc.store.ReactNativeFetchResponse.Error
+import org.wordpress.android.fluxc.store.ReactNativeFetchResponse.Success
+import org.wordpress.android.fluxc.store.ReactNativeStore
+import org.wordpress.android.modules.BG_THREAD
+import javax.inject.Inject
+import javax.inject.Named
+
+class ReactNativeRequestHandler @Inject constructor(
+    private val reactNativeStore: ReactNativeStore,
+    @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher
+) {
+    fun performGetRequest(
+        pathWithParams: String,
+        mSite: SiteModel,
+        onSuccess: Consumer<String>,
+        onError: Consumer<String>
+    ) {
+        GlobalScope.launch(bgDispatcher) {
```

Comment:
GlobalScope should generally not be used. We should add the `CoroutineScope` interface to this class and add hooks to `onCreate` and `onDestroy` where the first one initializes the scope and the second one cancels it. When I'm thinking about it, it's probably enough to just hook to `onDestroy` to cancel the job since the `ReactNativeRequestHandler` is always recreated with the activity. 

The disadvantage of the GlobalScope is that it creates a memory leaks (of the two passed Consumers). The operation is also not cancelled when the activity is destroyed.

Reply:
That makes sense. Thanks for explaining. I've updated this. I did implement this in a slightly different way from what I've seen elsewhere in the app (instead of storing the job in a field, I just got it from the `coroutineContext` when I needed to cancel it), so let me know if you see any issues with that since I don't have a lot of experience with coroutines.